### ISSUE TITLE
Gemini exchange trying to call Doc instead of Api -CORS issue

### ISSUE THREAD

**Issue by oesile** [ccxt/ccxt#8567]
Hi gusy,

I'm trying to use gemini as  exchanger but when I try to retrieve the balance get: blocked by CORS. I tried to use the proxy property with has.CORS = false/true and nothing work. The weird think is that the cols is to  https://**docs**.gemini.com/**rest-api** (?) instead of https://api.sandbox.gemini.com. So I guess something is wrong here.

- OS: Windows
- Programming Language version: Node.js (WEB)
- CCXT version: 1.42.67

```
this.gemini=new ccxt.gemini({ enableRateLimit: true,
    apiKey: 'xxxxxxxxxxxxxxxx',
    secret: 'xxxxxxxxxxxxxxxx',
});

  async getBalance(){
        this.data= await this.gemini.fetchBalance();
  }

getBalance();

```

```
Access to fetch at 'https://docs.gemini.com/rest-api' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Comment by kroitor** [comment]
> The weird think is that the cols is to https://docs.gemini.com/rest-api (?) instead of https://api.sandbox.gemini.com.

This is not weird. In fact, the exchanges' APIs are far from ideal. Sometimes they don't serve the info required to parse order limits from their APIs (Gemini does not serve that info). Therefore we had to add a web-api parser, that loads those limits from their docs, cause that's the only place to get that info. That explains the request going to the `docs`, upon loading the markets (which happens before fetchBalance() implicitly).

In general this works ok in Node, but not in the browser, where you have more strict in-browser security rules – you can't load external domains without resolving the CORS issue first.

>  I tried to use the proxy property with has.CORS = false/true and nothing work.

That is a wrong way of using the CORS proxy. The correct way to use a CORS proxy is explained here: https://github.com/ccxt/ccxt/wiki/Install#cors-access-control-allow-origin

1. you have to launch a cors proxy on your server or locally
2. you have to point ccxt to the cors using the `proxy` property as explained in the manual
3. then your in-browser requests will work

Let us know if the above does not answer your questions. Feel free to reopen this issue or just ask further questions if any.

**Comment by oesile** [comment]
> order limits

Hi kroitor,

Thanks for you quick response!

All clear with the use of doc and api, thanks for the explanation.

Regarding the cors. I did my due diligence before posting and I believe didn't express properly myself.   When I said "I tried" to use the proxy property and cors. I meant, I test it with a local server and cors services

this.gemini=new ccxt.gemini({ enableRateLimit: true,
    apiKey: 'xxxxxxxxxxxxxxxx',
    secret: 'xxxxxxxxxxxxxxxx',
   has: { CORS: false, }  //also try true 
});

//this.gemini.proxy = 'https://crossorigin.me/';
this.gemini.proxy = 'https://localhost:8080/';

in any case I get:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://crossorigin.me/https://docs.gemini.com/rest-api. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).

I notice that the request is sending always 'origin' on the header that means is a CORS request. Independently of the value in has:{ CORS: true/false} . I take the request from ccxt and resending removing the 'origin' and get 200 status.

With origin:
![image](https://user-images.githubusercontent.com/6927502/110611085-2b32fb80-818f-11eb-9730-714bbda4da3f.png)

Same request but without origin:
![image](https://user-images.githubusercontent.com/6927502/110610917-faeb5d00-818e-11eb-9962-d69059225e4f.png)

**Comment by oesile** [comment]
Ok, no is clear that has.CORS, I read in another issue that you should combine has.CORS=false and the proxy.
My app is a NodeJS web app. So, I don't use ccxt.browser.js.
